lightning/api_clnraw.go

```diff
@@ -767,3 +742,288 @@ func (l *ClnRawLightningAPI) GetInternalChannelsAll(ctx context.Context) (map[st
 
 	return result, nil
 }
+
+// ConnectPeer - API call.
+func (l *ClnRawLightningAPI) ConnectPeer(ctx context.Context, id string) error {
+	var resp ClnConnectResp
+	err := l.connection.Call(ctx, Connect, []string{id}, &resp)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.ID) == 0 {
+		return fmt.Errorf("invalid response")
+	}
+
+	return nil
+}
+
+// GetOnChainAddress - API call.
+func (l *ClnRawLightningAPI) GetOnChainAddress(ctx context.Context) (string, error) {
+	var resp ClnNewAddrResp
+
+	err := l.connection.Call(ctx, NewAddr, []interface{}{}, &resp)
+	if err != nil {
+		return "", err
+	}
+
+	if len(resp.Bech32) == 0 {
+		return "", fmt.Errorf("invalid response")
+	}
+
+	return resp.Bech32, nil
+}
+
+// GetOnChainFunds - API call.
+func (l *ClnRawLightningAPI) GetOnChainFunds(ctx context.Context) (*Funds, error) {
+	const (
+		Confirmed   = "confirmed"
+		Unconfirmed = "uncomfirmed"
+		Mili        = 1000
+	)
+
+	var fundsReply ClnFundsChainResp
+	err := l.connection.Call(ctx, ListFunds, []string{}, &fundsReply)
+	if err != nil {
+		return nil, err
+	}
+
+	f := &Funds{}
+	f.ConfirmedBalance = 0
+	f.TotalBalance = 0
+	f.LockedBalance = 0
+
+	for _, one := range fundsReply.Outputs {
+		amount := int64(ConvertAmount(one.AmountMsat))
+		if !one.Reserved && one.Status == Confirmed {
+			f.ConfirmedBalance += (amount / Mili)
+			f.TotalBalance += (amount / Mili)
+		} else if !one.Reserved && one.Status == Unconfirmed {
+			f.TotalBalance += (amount / Mili)
+		} else if one.Reserved && (one.Status == Unconfirmed || one.Status == Confirmed) {
+			f.LockedBalance += (amount / Mili)
+		}
+	}
+
+	return f, nil
+}
+
+// SendToOnChainAddress - API call.
+func (l *ClnRawLightningAPI) SendToOnChainAddress(ctx context.Context, address string, sats int64, useUnconfirmed bool, urgency Urgency) (string, error) {
+	var reply ClnWithdrawResp
+
+	feerate := "normal"
+	switch urgency {
+	case Urgent:
+		feerate = "urgent"
+	case Normal:
+		feerate = "normal"
+	case Low:
+		feerate = "slow" // slow not low, beware!
+	}
+
+	minConf := 1
```

Comment:
minConf is the number of confirmations your UTXOs needs to have to be used. I don't think it makes sense to use something > 1 since this is for sending. It makes sense to distinguish between confirmed balance and unconfirmed balance. If you want to use the second it is more likely to fail. But if you use a confirmed UTXO.. well there could be a reorg and your balance will change and tx fail (but this is more of a rare corner-case) and tx might fail for other reasons too.